spidermon/utils/settings.py

```diff
@@ -15,3 +16,26 @@ def getdictorlist(crawler, name, default=None):
         except ValueError:
             return value.split(",")
     return copy.deepcopy(value)
+
+
+def get_aws_credentials(settings):
+    aws_access_key_id = settings.get("SPIDERMON_AWS_ACCESS_KEY")
+    aws_secret_access_key = settings.get("SPIDERMON_AWS_SECRET_KEY")
+
+    if aws_access_key_id and aws_secret_access_key:
+        warnings.warn(
+            "SPIDERMON_AWS_ACCESS_KEY and SPIDERMON_AWS_SECRET_KEY are deprecated. "
+            "Please update them to SPIDERMON_AWS_ACCESS_KEY_ID and SPIDERMON_AWS_SECRET_ACCESS_KEY. "  # noqa
+            "Scrapy settings AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are also valid.",
+            DeprecationWarning,
+        )
+
+    if not aws_access_key_id and not aws_secret_access_key:
```

Comment:
What if someone mix the settings names? `SPIDERMON_AWS_ACCESS_KEY` and `SPIDERMON_AWS_SECRET_ACCESS_KEY` for example? I believe it would be better if we handle each setting separately.

Reply:
Actually, I prefer them in pairs.
This is makes things clearer.
The case you mentioned, you are mixing the old style with the new one, so you should prefer the new one to avoid problems in the future.
In case of `SPIDERMON_AWS_SECRET_ACCESS_KEY` and `AWS_SECRET_ACCESS_KEY` I don't think it would happen because for scrapy you need to set both `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID`.
So, I think we can keep some consistency and the error happens earlier (because it won't work) instead of lost warning message.